scubainit/src/util.rs

```diff
@@ -75,6 +75,20 @@ pub fn pop_env_uint(name: &str) -> Result<Option<u32>> {
     Ok(Some(value))
 }
 
+/// Gets and unsets an octal-encoded uint environment variable `name`, or Ok(None) if it is not set.
+///
+/// # Errors
+///
+/// This function will return an error if the variable is set, but is not a valid integer string.
+pub fn pop_env_octal(name: &str) -> Result<Option<u32>> {
+    let value_str = match pop_env_str(name) {
+        None => return Ok(None),
+        Some(s) => s,
+    };
+    let value = u32::from_str_radix(&value_str, 8)
+        .context(format!("Parsing integer variable {name}=\"{value_str}\""))?;
```

Comment:
```suggestion
        .with_context(|| format!("Parsing integer variable {name}=\"{value_str}\""))?;
```

This way the string will only be allocated when the error actually occurs instead of being eagerly evaluated since it's directly a parameter to the method.